packages/react/counter/src/Counter.tsx

```diff
@@ -155,14 +158,24 @@ export function Counter(props: CounterProps) {
     setup()
   }, [fetchData, isStatic, query?.metric, value])
 
-  if (isLoading || loading) return <Loader styles={styles} />
-
-  if (hasError) {
+  if (hasError || propsMismatch) {
     return <ErrorFallback styles={styles} />
   }
 
+  if ((isLoading || loading || (!dataValue && !isStatic)) && !counterRef.current) {
```

Comment:
@felipecadavid please help review this line. Based on your comment yesterday, there seems to be some difference when Counter returns `null` versus `undefined`.

TBH, I would be very curious if we can get rid of that distinction. It feels too subtle, IMO.

Reply:
In order to document this behavior and make sure we don't break anything, I think we should write some tests.

We could do this in a followup PR. @felipecadavid perhaps we could check in one simple test that uses the same modular msw + react-testing-library approach we are using for our Console code? This would serve as a blueprint. Then, we can say that, for all follow-up PRs, we should write tests, following the [testing "trophy"](https://kentcdodds.com/blog/the-testing-trophy-and-testing-classifications) model. WDYT @sashathor?

Reply:
Yes, we currently make a distinction between `null` and `undefined`

`undefined` => No value was received from the API, should fallback in error.

`null` => Valid value from the API, should be handled and will show a `-`.

with this, a `null` value will be stuck in loading state.

Maybe we can think about a better way to handle this behavior, but for now I think it is a good chance to kickoff testing.

```suggestion
  const isLoadingState = isLoading || loading
  const isFetchedValue = dataValue === undefined && !isStatic

  if ((isLoadingState || isFetchedValue) && !counterRef.current) {
    return <Loader styles={styles} />
  }
```

Reply:
@markandrus I've added a new [task for Unit tests](https://linear.app/propel/issue/PRO-2131/introduce-unit-tests)

Reply:
@felipecadavid I suggest refactoring the logic around Loader during the merging of the components into one ui-kit lib.